p2p/node/peerManager/peerManager.go

```diff
@@ -369,27 +370,26 @@ func (pm *BasicPeerManager) getBestPeersWithFallback(location common.Location) [
 	bestPeersCount := pm.peerDBs[locName][Best].GetPeerCount()
 	if bestPeersCount < C_peerCount {
 		bestPeerList := pm.getPeersHelper(pm.peerDBs[locName][Best], bestPeersCount)
-		bestPeerList = append(bestPeerList, pm.getResponsivePeersWithFallback(location)...)
+		maps.Copy(bestPeerList, pm.getResponsivePeersWithFallback(location))
 		return bestPeerList
```

Comment:
I think we should check the length here before we return it, and do the lookups again

Reply:
its not that easy. This PR just gets deduplication. Filling out to a length will require a bit of redesign throughout, but it is not high priority*. This PR is to get an immediate fix for deduplication

* occasionally getting fewer peers than we requested (because of duplicated entries) is probabilistic. Usually you will still get enough peers to fill requests, so node behavior should still work.

Reply:
https://github.com/dominant-strategies/go-quai/issues/1740